src/Graph/Nodes/BinaryNode.php

```diff
@@ -16,70 +14,9 @@
 interface BinaryNode extends Node
 {
     /**
-     * Return the children of this node in a generator.
-     *
-     * @return \Traversable<\Rubix\ML\Graph\Nodes\BinaryNode>
-     */
-    public function children() : Traversable;
-
-    /**
-     * Return the left child node.
-     *
-     * @return \Rubix\ML\Graph\Nodes\BinaryNode|null
-     */
-    public function left() : ?BinaryNode;
```

Comment:
What's wrong about keeping left() / right() nodes for a BinaryNode...?

Reply:
Certain `BinaryNode`s such as leaf nodes do not have any children, by definition. The BinaryNode interface now reflects this. We have a separate interface `HasBinaryChildren` for BinaryNodes that *do* have children.